File: frontend/src/app/chess-game/model/MovePieceRequest.ts
import { Piece } from "./PieceInfo"

export interface MovePieceRequest {
    boardId: string
    startSquare: string
    targetSquare: string
    pawnPromotedToPieceType?: Piece.QUEEN | Piece.KNIGHT | Piece.ROOK | Piece.BISHOP,
    username: string
}

File: illchess-web-socket-service/illchess/illchess-adapter/src/main/java/pl/illchess/adapter/board/command/in/rest/dto/MovePieceRequest.java
package pl.illchess.adapter.board.command.in.rest.dto;

import pl.illchess.application.board.command.in.MovePieceUseCase;

import java.util.UUID;

public record MovePieceRequest(
    UUID boardId,
    String startSquare,
    String targetSquare,
    String pawnPromotedToPieceType,
    String username
) {
    public MovePieceUseCase.MovePieceCmd toCmd() {
        return new MovePieceUseCase.MovePieceCmd(
            boardId,
            startSquare,
            targetSquare,
            pawnPromotedToPieceType,
            username
        );
    }
}
